src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ComponentDesigner.cs

```diff
@@ -294,13 +294,17 @@ ICollection ITreeDesigner.Children
             get
             {
                 ICollection comps = AssociatedComponents;
-                IDesignerHost host = (IDesignerHost)GetService(typeof(IDesignerHost));
-                if (comps.Count > 0 && host != null)
+                if (comps != null && comps.Count > 0 && GetService(typeof(IDesignerHost)) is IDesignerHost host)
                 {
                     IDesigner[] designers = new IDesigner[comps.Count];
                     int idx = 0;
-                    foreach (IComponent comp in comps)
+                    foreach (object associatedObject in comps)
```

Comment:
Personally I'm inclined to use Linq here:
```suggestion
                    foreach (IComponent comp in comps.OfType<IComponent>())
```

`OfTypeIterator` is doing the same work, but makes the callsite simpler
https://github.com/dotnet/corefx/blob/8750960d3fafa46a9b838c351e995a01fa8b599f/src/System.Linq/src/System/Linq/Cast.cs#L22-L31

The same can be applied in few instances below